mono/metadata/monitor.c

```diff
@@ -333,7 +333,7 @@ mono_locks_dump (gboolean include_untaken)
 					to_recycle++;
 			} else {
 				if (!monitor_is_on_freelist ((MonoThreadsSync *)mon->data)) {
-					MonoObject *holder = (MonoObject *)mono_gchandle_get_target ((guint32)mon->data);
+					MonoObject *holder = (MonoObject *)mono_gchandle_get_target ((guint32)(size_t)mon->data);
```

Comment:
I'm not sure what's the story of the `g*` types, but I think we shouldn't mix them at least. So I would suggest `gsize` instead.